File: 019/019.py
day_of_year = 365 % 7
sundays = 0
months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
for year in range(1901, 2001):
    for i in range(12):
        number_of_days = months[i]
        if year % 4 == 0 and i == 1:
            number_of_days = 29
        day = day_of_year % 7
        day_of_year += number_of_days
        if day == 6:
            sundays += 1

print(sundays)

File: 019/019.c
#include <stdio.h>

int main() {
    int dayOfYear = 365 % 7;
    int sundays = 0;
    int months[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    for (int year = 1901; year < 2001; year++) {
        for (int i = 0; i < 12; i++) {
            int numberOfDays = months[i];
            if (year % 4 == 0 && i == 1)
                numberOfDays = 29;
            int day = dayOfYear % 7;
            dayOfYear += numberOfDays;
            if (day == 6)
                sundays++;
        }
    }
    printf("%d\n", sundays);
}
